### ISSUE TITLE
Better support for tag helpers that create new elements based on Razor source

### ISSUE THREAD

**Issue by dougbu** [aspnet/Razor#315]
When tag helpers create new elements primarily based on the targeted element from the Razor source, they need to completely take over rendering.  The MVC `<link/>` and `<script/>` helpers do this today.  Suggest extending `TagHelperOutput` so that it better supports these scenarios.  Potential changes include:
- Support adding individual attributes with values that are not HTML encoded when `StartTag()` is called
- Provide a method (an extension method?) which appends the entire element to another `TagHelperOutput` instance
- Make the constructor that does not require an `attributes` dictionary `public`
- Maybe provide a convenience method (an extension method?) which HTML encodes content then sets `Content`.  This would be similar to `TagBuilder.SetInnerText()`

TL;DR

`TagBuilder` covers some of the requirements `TagHelperOutput` does not.  But it does not work well in the above scenario because it does not preserve attribute order.

Other cases:
Tag helpers that deal with entire elements often create new elements where the majority of the generated code comes from the helper e.g. the MVC `<input/>` tag helper.  Authors are likely to reach for `TagBuilder` in these scenarios.

Tag helpers that make focused changes to the existing element need only make changes to the `TagHelperOutput` parameter passed to `Process()` or `ProcessAsync()`.

**Comment by Eilon** [comment]
Closing due to lack of demand.
